front-end/e-learning/src/components/quizzes/Question.js

```diff
@@ -0,0 +1,225 @@
+import React, { useState, useEffect } from "react";
+import { connect } from "react-redux";
+import { useNavigate, useParams } from "react-router-dom";
+// import { useCookies } from "react-cookie";
+
+import Button from "@mui/material/Button";
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import CardActions from "@mui/material/CardActions";
+import Typography from "@mui/material/Typography";
+import Grid from "@mui/material/Grid";
+import ArrowForwardIosRoundedIcon from "@mui/icons-material/ArrowForwardIosRounded";
+import Tooltip from "@mui/material/Tooltip";
+import { styled } from "@mui/material/styles";
+import IconButton from "@mui/material/IconButton";
+import ArrowBackIosRoundedIcon from "@mui/icons-material/ArrowBackIosRounded";
+import Box from "@mui/material/Box";
+import LinearProgress from "@mui/material/LinearProgress";
+
+import { listWord, saveAnswer } from "../../actions";
+
+const Question = ({
+  listWord,
+  words_choices,
+  saveAnswer,
+  data,
+  onSetOpen,
+  onSetState,
+  token,
+}) => {
+  // const [cookies, setCookie] = useCookies(["user"]);
+  const [currentWord, setCurrentWord] = useState(0);
+  const [answers, setAnswers] = useState([]);
+  const [count, setCount] = useState(0);
+
+  // let { id } = useParams();
+  let navigate = useNavigate();
+
+  const StyledNextIcon = styled(ArrowForwardIosRoundedIcon)(({ theme }) => ({
+    color: "#BB6464",
+  }));
+
+  const StyledPrevIcon = styled(ArrowBackIosRoundedIcon)(({ theme }) => ({
+    color: "#BB6464",
+  }));
+
+  const onAnswer = (word_id, choice_id) => {
+    if (answers[word_id] === undefined) setCount(count + 1);
+    saveAnswer(data.id, word_id, choice_id, token);
+    setAnswers({ ...answers, [word_id]: choice_id });
+    if (words_choices.length != currentWord + 1)
+      setCurrentWord(currentWord + 1);
```

Comment:
please put `{}` on if blocks so we wont get confused in the future as to which lines are included inside the if and which arent.